dcc-portal-api/src/main/java/org/icgc/dcc/portal/util/Filters.java

```diff
@@ -106,14 +106,18 @@ public static ObjectNode pathwayFilter() {
 
   public static ObjectNode inputGeneSetIdFilter(@NonNull String inputGeneSetId) {
     val geneFilter = geneFilter();
-    geneFilter.with("gene").set(API_ENTITY_SET_ID_FIELD_NAME, is(inputGeneSetId));
+    geneFilter.with("gene").set("id", is("ES:" + inputGeneSetId));
 
     return geneFilter;
   }
 
   public static ObjectNode inputGeneSetFilter(@NonNull UUID inputGeneSetId) {
+    // Method appears to be used by multiple contexts, with and without `ES:`
+    val setId = inputGeneSetId.toString();
+    val id = setId.contains("ES:") ? setId : ENTITY_SET_PREFIX + setId;
```

Comment:
Why not use `ENTITY_SET_PREFIX`  instead of `"ES:"`? Why not use `startsWith` instead of `constains`?